### ISSUE TITLE
range not rendering properly on safari

### ISSUE THREAD

**Issue by dvic** [petalframework/petal_components#357]
https://www.whatsmybrowser.org/b/381QC

The coloured area of the select range is not visible.

safari:
![image](https://github.com/user-attachments/assets/41a32bcc-940e-4746-9d4b-4c22f17039de)

chrome:
![image](https://github.com/user-attachments/assets/f16d05b0-84c1-4dae-809f-72024825a8b2)

**Comment by nhobes** [comment]
You're right - thanks for pointing this out. I'll look into it and come back to you.

**Comment by nhobes** [comment]
So I've done a fair bit of digging into this and attempted multiple solutions, but it's still giving me grief (so that the range still provides a dynamic fill without requiring JS).

I looked into [this link](https://codepen.io/ShadowShahriar/pen/zYPPYrQ), but keep getting inconsistent results with the thumb.

Long story short, there's a proposal to standardise input[type="range"] styling: https://github.com/w3c/csswg-drafts/issues/4410

I don't have time at the moment to investigate this further, but feel free to submit a PR if you come up with a solution :)

**Comment by nhobes** [comment]
For the record, this is where I got to, the dynamic fill is using a `clip-path` and `box-shadow` to simulate the progress fill.

![CleanShot 2024-10-11 at 14 38 11@2x](https://github.com/user-attachments/assets/9e3a2d19-b568-4d9e-92b1-931f6fcccb07)

**Comment by dvic** [comment]
> I looked into [this link](https://codepen.io/ShadowShahriar/pen/zYPPYrQ), but keep getting inconsistent results with the thumb.

What inconsistency are you referring to? On my iPhone 14 that link works just fine, haven't tried my Mac yet.

**Comment by nhobes** [comment]
Sorry, I just meant when I tried to modify our range with those styles, it didn't quite work out well cross-browser like I'd planned. I got close as you can see above with the screenshot I attached, but couldn't quite get the desired look and feel.

**Comment by dvic** [comment]
> Sorry, I just meant when I tried to modify our range with those styles, it didn't quite work out well cross-browser like I'd planned. I got close as you can see above with the screenshot I attached, but couldn't quite get the desired look and feel.

Ah clear

**Comment by mplatts** [comment]
Seems to work in safari for me.
